backend/coins/eth/account.go

```diff
@@ -812,3 +814,133 @@ func (account *Account) VerifyAddress(addressID string) (bool, error) {
+func (account *Account) SignTypedMsg(
+	chainId uint64,
+	data string,
+) (string, error) {
+	signedMessage, err := account.Config().Keystore.SignETHTypedMessage(chainId, []byte(data), account.signingConfiguration.AbsoluteKeypath())
+	if err != nil {
+		return "", err
+	}
+	return "0x" + hex.EncodeToString(signedMessage), nil
+}
+
+// TX proposal arguments received from Wallet Connect with Gas, GasPrice, Value and Nonce being optional
+type WalletConnectArgs struct {
+	From     string `json:"from"`
+	To       string `json:"to"`
+	Data     string `json:"data"`
+	Gas      string `json:"gas,omitempty"`
+	GasPrice string `json:"gasPrice,omitempty"`
+	Value    string `json:"value,omitempty"`
+	Nonce    string `json:"nonce,omitempty"`
+}
+
+// EthSignTypedMsg signs an Ethereum Tx received from WalletConnect
+func (account *Account) EthSignWalletConnectTx(
+	// send: whether transaction should be broadcast after signing
+	send bool,
+	// chainId: allow specifying other IDs than 1 (ETH mainnet) for other EVM networks
+	chainId uint64,
+	proposedTx WalletConnectArgs,
+) (string, string, error) {
+	var nonce uint64
+	var message ethereum.CallMsg
+	var gasPrice *big.Int
+	var value *big.Int
+
+	if !ethcommon.IsHexAddress(proposedTx.To) {
+		return "", "", errp.WithStack(errors.ErrInvalidAddress)
+	}
+	address := ethcommon.HexToAddress(proposedTx.To)
+	if isMixedCase(proposedTx.To) && proposedTx.To != address.Hex() {
+		return "", "", errp.WithStack(errors.ErrInvalidAddress)
+	}
+
+	if proposedTx.Nonce != "" {
+		parsed, err := strconv.ParseUint(strings.TrimPrefix(proposedTx.Nonce, "0x"), 16, 64)
+		if err != nil {
+			return "", "", err
+		}
+		nonce = parsed
+	} else {
+		nonce = account.nextNonce
+	}
+
+	if proposedTx.Value != "" {
+		bigIntValue := new(big.Int)
+		bigIntValue.SetString(strings.TrimPrefix(proposedTx.Value, "0x"), 16)
+		value = bigIntValue
+	}
+
+	data, err := hex.DecodeString(strings.TrimPrefix(proposedTx.Data, "0x"))
+	if err != nil {
+		return "", "", err
+	}
+
+	message = ethereum.CallMsg{
+		From:     account.address.Address,
+		To:       &address,
+		Gas:      0,
+		GasPrice: big.NewInt(0),
+		Value:    value,
+		Data:     data,
+	}
+
+	gasLimit, err := account.coin.client.EstimateGas(context.TODO(), message)
+	if err != nil {
+		if strings.Contains(err.Error(), etherscan.ERC20GasErr) {
+			return "", "", errp.WithStack(errors.ErrInsufficientFunds)
+		}
+		account.log.WithError(err).Error("Could not estimate the gas limit.")
+		return "", "", errp.WithStack(errors.TxValidationError(err.Error()))
+	}
+
+	for _, t := range account.feeTargets() {
+		//TODO Let user choose gas price/priority
+		if t.code == accounts.FeeTargetCodeNormal {
+			if t.gasPrice.Cmp(big.NewInt(0)) <= 0 {
+				return "", "", errors.ErrFeeTooLow
+			}
+			gasPrice = t.gasPrice
+		}
+	}
+
+	tx := types.NewTransaction(nonce,
+		*message.To,
+		message.Value, gasLimit, gasPrice, message.Data)
+	//TODO edit signer to match chainID proposed by wallet connect
```

Comment:
L#934 the signer config *params.ChainConfig is made with account.coin.Net() where the ChainID on the ChainConfig will I believe only be 1 currently in the BBApp. 
Which I think will not work with other networks right now because of tx replay 

Not sure what the best approach is here, copy the *params.ChainConfig from account.coin.Net() and keep it the same except for overwriting the ChainID?

Expanded the TODO

Reply:
so signing a tx currently does not work for non-ETH chains?

if so, i'd check the passed chainID and error out early if it is not equal to `account.coin.ChainID(). and worry about how to fix it in a future PR.